subsys/net/ip/tcp2.c

```diff
@@ -1029,6 +1029,74 @@ static struct tp *json_to_tp(void *data, size_t data_len)
 	return &tp;
 }
 
+struct tp_msg {
+	const char *msg;
+};
+
+static const struct json_obj_descr tp_msg_dsc[] = {
+	JSON_OBJ_DESCR_PRIM(struct tp_msg, msg, JSON_TOK_STRING),
+};
+
+static enum tp_type json_decode_msg_type(void *data, size_t data_len)
+{
+	struct tp_msg tp;
+
+	memset(&tp, 0, sizeof(tp));
+
+	json_obj_parse(data, data_len, tp_msg_dsc, ARRAY_SIZE(tp_msg_dsc), &tp);
```

Comment:
The return value for `json_obj_parse()` is an integer. The nth least significant bit is set if the nth field has been correctly decoded; in this case, if `json_obj_parse & BIT(0)`, `tp.msg` has an actual value.

Reply:
@lpereira 

OK. Thanks a lot!